backend/src/auth/credentials.ts

```diff
@@ -0,0 +1,80 @@
+import { Buffer } from 'buffer';
+import { pbkdf } from 'bcrypt-pbkdf';
+import * as jwt from 'jsonwebtoken';
+import config from '../config';
+
+/* Checks if a password matches a provided hash, returning true if it did.
+ * 'password' *must* be a utf-8 encoded string, whilst 'existing_hash' *must* be an ascii encoded string.
+ */
+export const checkPassword = (password: string, existing_hash: string) => {
+  // This is *meant* to be slow. In fact, the slowness is built into the algorithm.
+  // The reason this is slow is because it is used in order to reduce the efficacy of
+  // brute force attacks. The slowness is controlled by the 'rounds' parameter.
+
+  const decoded_hash = decodePasswordHash(existing_hash);
```

Comment:
Snake case in TypeScript will be a problem later!

Reply:
Thanks for catching that. It seems like I'm just too used to snake case...

Reply:
We need to relax some of the lint-rules, though. I need bitwise operations in order to implement password comparisons securely.

Reply:
I'm doing a somewhat dirty per line fix for this by disabling the eslint rule on the lines that use bitwise ops, which might be enough.

Reply:
It should be fixed now, as eslint passes without error on the backend for me. I haven't touched the backend, which is why the CI fails.